L1Trigger/DemonstratorTools/src/codecs_htsums.cc

```diff
@@ -6,11 +6,12 @@ namespace l1t::demo::codecs {
   ap_uint<64> encodeHtSum(const l1t::EtSum& htSum) {
     l1tmhtemu::EtMiss htMiss;
     htMiss.Et = htSum.p4().energy();
+    ap_uint<l1tmhtemu::kMHTSize> ET = htMiss.Et.to_uint();
     htMiss.Phi = htSum.hwPhi();
-    l1tmhtemu::Et_t HT = htSum.hwPt();
+    ap_uint<l1tmhtemu::kMHTSize> HT = htSum.hwPt();
```

Comment:
If you're trying to limit the word to the 16 LSBs, then you should define the type in L1TkHTMissEmulatorProducer.h.